src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -147,9 +153,14 @@ protected void configureSchemalessOutput(BatchSinkContext context,
     String splitField = config.getSplitField();
     String projectName = config.getDatasetProject();
     String datasetName = config.getDataset();
-    context.addOutput(Output.of(config.getReferenceName(),
-                                new DelegatingMultiSinkOutputFormatProvider(conf, splitField, bucket,
-                                                                            projectName, datasetName)));
+    String fqn = BigQueryUtil.getFQN(config.getDatasetProject(),
+                                     config.getDataset(), config.getTable());
+    String referenceName = Strings.isNullOrEmpty(config.getReferenceName())
+      ? ReferenceNames.normalizeFqn(fqn)
```

Comment:
There's a possibility of having clashes using this normalizing method defined [here](https://github.com/cdapio/hydrator-plugins/blob/62cee359718523171d522a0327aa7713d0a257d7/hydrator-common/src/main/java/io/cdap/plugin/common/ReferenceNames.java#L65).

We could have a case of `a://b` and `a:b` which both be normalized to `a.b`. Wdyt @albertshau @yaojiefeng ?

Reply:
that's fine. It is expected that if no reference name is given, the auto-generated name can have conflicts, there's no way to avoid it.